saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/AbstractSaml2AuthenticationRequest.java

```diff
@@ -46,20 +47,26 @@ public abstract class AbstractSaml2AuthenticationRequest implements Serializable
 
 	private final String authenticationRequestUri;
 
+	private final String relyingPartyRegistrationId;
+
 	/**
 	 * Mandatory constructor for the {@link AbstractSaml2AuthenticationRequest}
 	 * @param samlRequest - the SAMLRequest XML data, SAML encoded, cannot be empty or
 	 * null
 	 * @param relayState - RelayState value that accompanies the request, may be null
 	 * @param authenticationRequestUri - The authenticationRequestUri, a URL, where to
 	 * send the XML message, cannot be empty or null
+	 * @param relyingPartyRegistrationId the registration id of the relying party
 	 */
-	AbstractSaml2AuthenticationRequest(String samlRequest, String relayState, String authenticationRequestUri) {
+	AbstractSaml2AuthenticationRequest(String samlRequest, String relayState, String authenticationRequestUri,
+			String relyingPartyRegistrationId) {
 		Assert.hasText(samlRequest, "samlRequest cannot be null or empty");
 		Assert.hasText(authenticationRequestUri, "authenticationRequestUri cannot be null or empty");
+		Assert.hasText(relyingPartyRegistrationId, "relyingPartyRegistrationId cannot be null or empty");
```

Comment:
Let's remove this so that payloads Jackson-serialized in 5.7 can be deserialized in 5.8. We can add a ticket to require this in Spring Security 6 where we can make a breaking change.